src/cookbooks/data_modeling/aggregates_guidelines.md

```diff
@@ -0,0 +1,91 @@
+  [Cookbook link to create an aggregate table](https://mozilla.github.io/bigquery-etl/cookbooks/common_workflows/#adding-a-new-scheduled-query).
+1. BigQuery Materialized Views. These are views defined by the developer using only SELECT and created, managed and scheduled by BigQuery. BigQuery creates aggregate tables to store the results of the view definition and thus optimize the queries.
+  [TODO: Instructions to implement a Materialized view]
+  ```
+  CREATE MATERIALIZED VIEW IF NOT EXISTS <dataset>.<materialized_view_name>
+  PARTITION BY
+    [partition_fields]
+  CLUSTER BY
+    [cluster_fields]
+  OPTIONS
+    (enable_refresh = TRUE, refresh_interval_minutes = 720)
+  AS
+  WITH <subquery_name> AS ( <subquery> ),
+  SELECT
+    <[columns]>
+  FROM
+    <subquery_name>
+  WHERE
+    <partition_field> > TIMESTAMP(<starting_date_to_backfill_the_view>)
+  GROUP BY
+    <[fields]>
+  ```
+  - An example of a materialized view definition.
+
+    1. Looker PDTs & aggregate awareness. These are aggregations that a developer defines in an Explore file (explore.lkml). From this definition, Looker creates a table in BigQuery's `mozdata.temp` using the naming convention`scratch schema + table status code + hash value + view name` and runs the scheduled update of the data.
+       Looker's PDTs and aggregate awareness are only used when at least one of the columns defined in the PDT are used in a Looker object and can be particularly beneficial to avoid having to rebuild the dashboards after a schema change. 
+       - This is a template that can be used to define an aggregate in a Looker Explore:
+       ```
+       aggregate_table: <aggregate_name: Descriptive name of this aggregation.> {
+        query: {
+        dimensions: [<table>.<columns>]
+        measures: [<table>.<columns>]
+        filters: [<table>.<partition_column>: "<Period to materialize in the aggregate e.g. 2 years>"]
+        }
+        materialization: {
+        sql_trigger_value: SELECT CURRENT_DATE() ;;
+        increment_key: <table>.<partition_column>
+        increment_offset: <INT: number of periods to update, recommended is 1.> }
+        }
+       ```
+       - An [Example of aggregate awareness in a Looker Explore](https://mozilla.cloud.looker.com/projects/spoke-default/files/combined_browser_metrics/explores/active_users_aggregates.explore.lkml)
+
+## Important considerations:
+
+- [Store the business logic in BigQuery](https://docs.telemetry.mozilla.org/cookbooks/data_modeling/where_to_store), ideally in a client_id level table to aggregate from.
+- All aggregates are version controlled in the git repositories `bigquery-etl` and `spoke-default`.
+- All types of aggregates require a backfill when the source data changes.
+  - The managed-backfill process can be used for aggregate tables.
+  - The materialized views are backfilled by ?. Any schema change in the base table invalidates the view and requires backfill.
```

Comment:
```suggestion
  - Materialized views cannot be backfilled, instead a materialized view needs to be re-created. Schema changes in base tables invalidates the view and requires it to be re-created. Materialized views scan all historical data of their referenced base tables by default, so **ensure to set a date filter to reduce the amount of data to be scanned**.
  - Add a date filter to materialized views to limit the amount of data to be scanned when these views get deployed initially. Otherwise they will scan the entire data in referenced base tables.
```